File: src/offer10-4.rectCover/rectCover.cpp
class Solution {
public:
    int rectCover(int number) {
        if (number <= 2) return number;
        int p = 1;
        int q = 2;
        for(int i = 3; i <= number; i++){
            q = q + p;
            p = q - p;
        }
        return q;
    }
};

File: src/offer10-4.rectCover/rectCover.java
public class Solution {
    public int RectCover(int target) {
        int p = 1;
        int q = 2;
        if(target <= 2){
            return target;
        }
        for(int i = 3; i <= target; i++){
            q += p;
            p = q - p;
        }
        return q;
    }
}